lib/teleterm/clusters/cluster.go

```diff
@@ -54,33 +54,67 @@ type Cluster struct {
 	// Auth server features
 	// only present where the auth client can be queried
 	// and set with GetClusterFeatures
-	Features *proto.Features
+	Features     *proto.Features
+	LoggedInUser LoggedInUser
 }
 
 // Connected indicates if connection to the cluster can be established
 func (c *Cluster) Connected() bool {
 	return c.status.Name != "" && !c.status.IsExpired(c.clock)
 }
 
-// GetClusterFeatures returns a list of features enabled/disabled by the auth server
-func (c *Cluster) GetClusterFeatures(ctx context.Context) (*proto.Features, error) {
-	var authPingResponse proto.PingResponse
+// EnrichCluster will make a network request to the auth server and add details to the
+// current Cluster that cannot be found on the disk only, including details about the LoggedInUser
+// and enabled enterprise features. This method requires a valid cert.
+func (c *Cluster) EnrichCluster(ctx context.Context) (*Cluster, error) {
+	var (
+		authClient   auth.ClientI
+		proxyClient  *client.ProxyClient
+		err          error
+		pingResponse proto.PingResponse
+		caps         *types.AccessCapabilities
+	)
```

Comment:
`pingResponse` and `caps` are the only two variables that need to survive outside of the callable in `addMetadataToRetryableError`, as far as I can tell.

Reply:
Good catch. Removed. Thanks!